plugins/pur/core/src/main/java/org/pentaho/di/repository/pur/PurRepository.java

```diff
@@ -506,11 +524,18 @@ public void deleteRepositoryDirectory( final RepositoryDirectoryInterface dir )
   public void deleteRepositoryDirectory( final RepositoryDirectoryInterface dir, final boolean deleteHomeDirectories )
     throws KettleException {
     try {
-      // Fetch the folder to be deleted
-      RepositoryFile folder = pur.getFileById( dir.getObjectId().getId() );
 
-      // Fetch the user's home directory
-      RepositoryFile homeFolder = pur.getFile( ClientRepositoryPaths.getUserHomeFolderPath( user.getLogin() ) );
+      readWriteLock.readLock().lock();
+      // Fetch the folder to be deleted
+      RepositoryFile folder;
+      RepositoryFile homeFolder;
+      try {
+        folder = pur.getFileById( dir.getObjectId().getId() );
+        // Fetch the user's home directory
+        homeFolder = pur.getFile( ClientRepositoryPaths.getUserHomeFolderPath( user.getLogin() ) );
+      } finally {
```

Comment:
The read lock should not be release between this and lower in the same method. The lock should be kept.